src/lib/components/tooltip/QTooltip.svelte

```diff
@@ -1,17 +1,183 @@
-<script lang="ts">
+<script lang="ts" generics="T extends HTMLElement | string">
+  import { mount, unmount, onMount, untrack } from "svelte";
+  import QTooltipBase from "./QTooltipBase.svelte";
   import type { QTooltipProps } from "./props";
 
-  let { value, position = "bottom", children, ...props }: QTooltipProps = $props();
+  let {
+    target,
+    value = $bindable(false),
+    position = "bottom",
+    offset = { x: 0, y: 0 },
+    delay = 50,
+    hideDelay = 50,
+    children,
+    ...props
+  }: QTooltipProps<T> = $props();
 
-  Q.classes("q-tooltip", {
-    bemClasses: {
-      active: value,
-      [position]: true,
-    },
-    classes: [props.class],
+  let tooltipHelperEl = $state<HTMLDivElement>();
+  let tooltipEl = $state<HTMLDivElement>();
+
+  let realTarget = $state<HTMLElement>();
+
+  let timerShow = $state<ReturnType<typeof setTimeout> | null>(null);
+  let timerHide = $state<ReturnType<typeof setTimeout> | null>(null);
+
+  let mountedTooltip: ReturnType<typeof mountTooltip> | null = null;
+
+  const id = Date.now();
+
+  $effect(() => {
+    if (value) {
+      untrack(show);
+    } else {
+      untrack(hide);
+    }
+  });
+
+  onMount(() => {
+    if (target) {
+      realTarget =
+        typeof target === "string"
+          ? document.querySelector<HTMLElement>(target) || undefined
+          : target;
+    } else {
+      let parent = tooltipHelperEl?.parentElement || null;
+
+      while (parent) {
+        if (parent.attributes.getNamedItem("data-quaff")) {
+          realTarget = parent;
+          break;
+        } else {
+          parent = parent.parentElement;
+        }
+      }
+    }
+
+    realTarget?.addEventListener("mouseenter", show);
```

Comment:
We have this repeating three times, maybe it would be good to have a function to reduce code duplication. For example, the function could take args `el: HTMLElement` and `addOrRemove: 'add' | 'remove'` (didn't check types).

Reply:
Yeah, thought about that. Will do it